pkg/master/controller.go

```diff
@@ -179,11 +179,15 @@ func createEndpointPortSpec(endpointPort int, endpointPortName string, extraEndp
 
 // CreateMasterServiceIfNeeded will create the specified service if it
 // doesn't already exist.
-func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP net.IP, servicePorts []api.ServicePort, serviceType api.ServiceType) error {
+func (c *Controller) CreateOrUpdateMasterServiceIfNeeded(serviceName string, serviceIP net.IP, servicePorts []api.ServicePort, serviceType api.ServiceType) error {
 	ctx := api.NewDefaultContext()
-	if _, err := c.ServiceRegistry.GetService(ctx, serviceName); err == nil {
+	if s, err := c.ServiceRegistry.GetService(ctx, serviceName); err == nil {
 		// The service already exists.
-		return nil
+		if svc, updated := updateMasterServiceIfNeeded(s, servicePorts, serviceType); updated {
+			glog.Warningf("Resetting spec for master service %q to %v", serviceName, svc)
```

Comment:
mm, I see the endpoints were already `Warningf`... don't feel strongly if we want to keep that